File: Contains_Duplicate.java
/**
 * Problem: https://leetcode.com/problems/contains-duplicate/
 * 
 * 1, Using HashSet to keep the record of every integer which has been visited.
 * O(n) in computing and space.
 * 
 * 
 */

public class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> set = new HashSet<Integer>();
        for(int x : nums) {
            if(set.contains(x)) return true;
            set.add(x);
        }
        return false;
    }
}

File: Swift/contains-duplicate.swift
/**
 * https://leetcode.com/problems/contains-duplicate/
 * 
 * 
 */ 
// Date: Sat May  2 12:45:51 PDT 2020
class Solution {
    func containsDuplicate(_ nums: [Int]) -> Bool {
        var set: Set<Int> = []
        for n in nums {
            if set.contains(n) {
                return true
            }
            set.insert(n)
        }
        return false
    }
}